puffin_egui/src/flamegraph.rs

```diff
@@ -731,10 +753,14 @@ fn paint_scope(
         }
 
         if result == PaintResult::Hovered {
+            let scope_details = info.scope_collection.fetch_by_id(&scope.id).unwrap();
             egui::show_tooltip_at_pointer(&info.ctx, Id::new("puffin_profiler_tooltip"), |ui| {
-                ui.monospace(format!("id:       {}", scope.record.id));
-                if !scope.record.location.is_empty() {
-                    ui.monospace(format!("location: {}", scope.record.location));
+                ui.monospace(format!("id:       {}", scope_details.function_name));
```

Comment:
mismatch between key (`id`) and value (`function_name`). This should probably be something like

```suggestion
                ui.monospace(format!("name:    {}", scope_details.name()));
```

where `ScopeDetails::name` either returns the named passed to `profile_scope!`, or the function name, or whatever is non-empty basically (e.g. for custom/user scopes)